app/controllers/api/v1/task_category_controller.rb

```diff
@@ -0,0 +1,51 @@
+class Api::V1::TaskCategoryController < ApplicationController
+  before_action :require_admin, only: [:index, :delete, :update, :create]
+
+  def index 
+    service = TaskCategoryService.new
+    list_all = service.find_all
+    if service.errors.present?
+      render json: { error: service.errors }, status: :unprocessable_entity
+    elsif list_all.blank?
+      render json: { message: 'Task Category List is Empty' }, status: :ok  
+    else
+      render json: list_all, each_serializer: TaskCategorySerializer, status: :ok
+    end
+  end
+
+  def delete
+    service = TaskCategoryService.new 
```

Comment:
a new object is being created in each action, think how this can be cached or moved to a common method